mlir/include/mlir/Bytecode/BytecodeImplementation.h

```diff
@@ -400,6 +400,10 @@ class DialectBytecodeWriter {
 
   /// Return the bytecode version being emitted for.
   virtual int64_t getBytecodeVersion() const = 0;
+
+  /// Retrieve the dialect version by name if available.
+  virtual FailureOr<const DialectVersion *>
+  getDialectVersion(StringRef dialectName) const = 0;
```

Comment:
suggestion: can you also add versions of this API that accept a `Dialect *`, or one like `getDialectVersion<MyDialect>()`? MLIR APIs typical expose both: strings for flexibility, and a template version for more idiomatic code.

Reply:
Thanks for taking a look. I added the second API - unfortunately it can't be a template though since it is a virtual method.

Reply:
Can't you write

```
template <typename DialectT>
FailureOr<const DialectVersion *>
getDialectVersion() const { return getDialectVersion(DialectT::getDialectNamespace()); }
```

Reply:
Sounds good, I think that one works :)